### PR TITLE
Implement puppy token page

### PR DESCRIPTION
**Pull request by tzulin**
## Description
Add a new page to display PUPPY token, and do operations to the token.

### functions
- [x] transfer
- [x] mint
- [x] burn
- [x] transferOwnership
> maybe won't implement these below, if you really want to do any of them, please let me know before tomorrow morning, thanks.
- [ ] approve
- [ ] transferFrom
- [ ] decreaseAllowance
- [ ] increaseAllowance

### UI
- [ ] make the token symbol look bigger

### CODE REVIEW

**Comment by Xanonymous-GitHub** [comment]
Screen Shot 2022-06-23 at 5 30 19 PM

Nice.

File: src/components/PuppyToken.jsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+      fill="none"
+      viewBox="0 0 24 24"
+    >
+      <circle
+        className="opacity-25"
+        cx="12"
+        cy="12"
+        r="10"
+        stroke="currentColor"
+        stroke-width="4"
```
Comment (Xanonymous-GitHub) [inline src/components/PuppyToken.jsx]:
```suggestion
        strokeWidth="4"
```

We should change it to camelCase when using Jsx.

Comment (Xanonymous-GitHub) [reply src/components/PuppyToken.jsx]: Addressed in [98d2d4b](https://github.com/lohsuan/puppy_sponsor/pull/14/commits/98d2d4bd70c5ed8f12af1e29d0f036c7ecd8f64a).

**Review by Xanonymous-GitHub — APPROVED** [review]
LGTM, Just make sure the file is formatted again before the final merge.

**Comment by Xanonymous-GitHub** [comment]
And seems I have caught the `transactionContext` on the `AddPuppy` page that repeatedly hit `updateChainContents`, it seems to be repeated once (totally hit twice), you could check it if you have time~
(and since the `transactionContext` is a global state, so it doesn't happen only on the page I describe)

**Review by lohsuan — APPROVED** [review]
LGTM:)

**Pull request merged**